common/utils.go

```diff
@@ -227,16 +227,25 @@ func GetBlockSubmissionInfo(submission *VersionedSubmitBlockRequest) (*BlockSubm
 	if err != nil {
 		return nil, err
 	}
-	// TODO (deneb): after deneb fork error if no blob fields
-	var (
-		blobs         []deneb.Blob
-		blobGasUsed   uint64
-		excessBlobGas uint64
-	)
-	if submission.Version == spec.DataVersionDeneb {
-		blobs = submission.Deneb.BlobsBundle.Blobs
-		blobGasUsed = submission.Deneb.ExecutionPayload.BlobGasUsed
-		excessBlobGas = submission.Deneb.ExecutionPayload.ExcessBlobGas
+	blobs, err := submission.Blobs()
+	if submission.Version >= spec.DataVersionDeneb && err != nil {
+		return nil, err
+	}
+	blobGasUsed, err := submission.BlobGasUsed()
+	if submission.Version >= spec.DataVersionDeneb && err != nil {
+		return nil, err
+	}
+	excessBlobGas, err := submission.ExcessBlobGas()
+	if submission.Version >= spec.DataVersionDeneb && err != nil {
+		return nil, err
+	}
+	depositReceipts, err := submission.DepositReceipts()
```

Comment:
this will error with unsupported version for deneb payloads. I suggest that you check the payload version for electra here.

Reply:
oh I see you check after calling this, disregard